app/controllers/Application.java

```diff
@@ -1689,6 +1698,168 @@ public static Result restFlowMetricsGraphData(String flowDefId) {
     return ok(new Gson().toJson(sortedDatasets));
   }
 
+  /* Rest Api implementation for providing the tuning parameters details for a job
+   * @param jobId - job execution id passed as a parameter for the GET request
+   * @result JSON containing suggested tuning paramter details
+   * eg: Formatted JSON Data
+    {
+      "tunein":{
+      "id":"https://ltx1-holdemaz01.grid.linkedin.com:8443/executor?execid=1234567&job=dimDataSimpleFlow_dimDataSimple&attempt=0",
+      "jobSuggestedParamSetId":1234,
+      "tuningAlgorithmId":2,
+      "autoApply":false,
+      "tuningAlgorithm":"HBT",
+      "tuningAlgorithms":[
+         {
+            "name":"HBT"
+         },
+         {
+            "name":"OBT"
+         }
+      ],
+      "iterationCount":10,
+      "tuningParameters":[
+         {
+            "name":"mapreduce.map.memory.mb",
+            "jobSuggestedParamValue":"1113.33",
+            "userSuggestedParamValue":"1113.33",
+            "currentParamValue":"1113.33"
+         },
+         {
+            "name":"mapreduce.map.java.opts",
+            "jobSuggestedParamValue":"839",
+            "userSuggestedParamValue":"839",
+            "currentParamValue":"839"
+         }
+      ]
+    }
+   }
+  */
+  public static Result getTuningParameter(String jobId) {
+    logger.info("Getting Tuning Parameters for " + jobId);
+    JsonObject parent = new JsonObject();
+    JsonObject tuneIn = new JsonObject();
+    try {
+      JobExecution jobExecution = JobExecution.find.select("*")
+          .where()
+          .eq(JobExecution.TABLE.jobExecId, jobId)
+          .findUnique();
+      Integer jobDefinitionId = 0;
+      if (jobExecution == null) {
+        logger.info("No Job Execution enrty found for job execution " + jobId);
```

Comment:
Shouldn't this be error and return from here as below code is meaningless to execute if you don't have execution context (jobdefinition id)

Reply:
Throwing an exception here.